Apps/Playground/UWP/App.cpp

```diff
@@ -165,15 +163,34 @@ concurrency::task<void> App::RestartRuntimeAsync(Windows::Foundation::Rect bound
         });
     });
 
+    // Initialize NativeWindow plugin.
+    size_t width = static_cast<size_t>(bounds.Width * m_displayScale);
+    size_t height = static_cast<size_t>(bounds.Height * m_displayScale);
+    auto* windowPtr = reinterpret_cast<ABI::Windows::UI::Core::ICoreWindow*>(CoreWindow::GetForCurrentThread());
+    m_runtime->Dispatch([windowPtr, width, height](Napi::Env env)
+    {
+        Babylon::NativeWindow::Initialize(env, windowPtr, width, height);
+    });
+
+    // Initialize NativeEngine plugin.
+    Babylon::InitializeNativeEngine(*m_runtime, windowPtr, width, height);
```

Comment:
I'm a bit confused why the windowPtr, width, and height need to be passed to both NativeWindow and NativeEngine.

Reply:
This, in part, is just me porting over what the existing API was rather than making too many invasive API changes at this point in the refactor. However, the reason the API was that way is because of macOS and iOS. Those three values are required (in the current implementation) for the initialization of bgfx, and my understanding is that the Apple operating systems require bgfx to be initialized on the main thread, where it doesn't have access to the `NativeWindow` because the `Napi::Env` can't be accessed on that thread and may not even exist yet anyway.

To be clear, this is something I'd definitely like to think about revising, but I'm trying to keep cleanup work on the plugins themselves to a minimum in this change. I'm currently hoping to limit this PR as much as possible to restructuring, after which we can tackle the individual plugins in relative isolation.